code/obj/item/gun/kinetic.dm

```diff
@@ -623,23 +623,69 @@ ABSTRACT_TYPE(/obj/item/gun/kinetic)
 	name = "Riot Shotgun"
 	desc = "A police-issue shotgun meant for suppressing riots."
 	icon = 'icons/obj/48x32.dmi'
-	icon_state = "shotty"
+	icon_state = "shotty-empty"
 	item_state = "shotty"
 	force = 15.0
 	contraband = 5
 	caliber = 0.72
 	max_ammo_capacity = 8
-	auto_eject = 1
+	auto_eject = 0
 	can_dual_wield = 0
 	two_handed = 1
 	has_empty_state = 1
 	gildable = 1
+	var/racked_slide = FALSE
 
 	New()
 		ammo = new/obj/item/ammo/bullets/abg
 		set_current_projectile(new/datum/projectile/bullet/abg)
 		..()
 
+	update_icon()
+		. = ..()
+		if(src.racked_slide)
+			src.icon_state = "shotty"
+		else
+			src.icon_state = "shotty-empty"
+
+	canshoot()
+		if (racked_slide)
+			return ..()
+		else
+			boutput(usr, "<span class='notice'>You need to rack the slide before you can fire!</span>") //Pain
+			return 0
+
+	shoot(var/target,var/start ,var/mob/user)
+		..()
+		src.racked_slide = FALSE
+		if (src.ammo.amount_left == 0) // change icon_state to empty if 0 shells left
+			src.update_icon()
+		src.casings_to_eject = 0 //This probably works
+
+	attack_self(mob/user as mob)
+		..()
+		if (!racked_slide) //Are we racked?
+			if (src.ammo.amount_left == 0)
+				boutput(user, "<span class ='notice'>You are out of shells!</span>")
+				icon_state = "shotty-empty"
+			else
+				racked_slide = TRUE
+				if (icon_state == "shotty") //"animated" racking
+					icon_state = "shotty-empty"
+					SPAWN_DBG(2) //FORGIVE ME ZEWAKA
+					icon_state = "shotty"
```

Comment:
did you even test this?

no is the likely answer because this shouldn't work, and if it does, it's a byproduct of lag not this code

Reply:
also read our code guide, this is in violation

Reply:
> 
> 
> did you even test this?
> 
> no is the likely answer because this shouldn't work, and if it does, it's a byproduct of lag not this code
> 
> look at other usages of this in the code

I did and it worked locally, so probably lag. 

I just removed it entirely and replaced it with animate() because I was informed animate exists.